File: api/src/APIBundle/Entity/Event.php
<?php

namespace APIBundle\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity()
 * @ORM\Table(name="events")
 */
class Event
{
    /**
     * @ORM\Id
     * @ORM\Column(type="string")
     * @ORM\GeneratedValue(strategy="UUID")
     */
    protected $event_id;

    /**
     * @ORM\Column(type="string")
     */
    protected $event_title;

    /**
     * @ORM\Column(type="string")
     */
    protected $event_description;

    /**
     * @ORM\ManyToOne(targetEntity="Exercise", inversedBy="exercise_events")
     * @ORM\JoinColumn(name="event_exercise", referencedColumnName="exercise_id", onDelete="CASCADE")
     * @var Exercise
     */
    protected $event_exercise;

    /**
     * @ORM\OneToMany(targetEntity="Incident", mappedBy="incident_event")
     * @var Incident[]
     */
    protected $event_incidents;

    /**
     * @ORM\ManyToOne(targetEntity="File")
     * @ORM\JoinColumn(name="event_image", referencedColumnName="file_id", onDelete="SET NULL", nullable=true)
     */
    protected $event_image;

    /**
     * @ORM\Column(type="smallint")
     */
    protected $event_order = 0;

    public function __construct()
    {
        $this->event_incidents = new ArrayCollection();
    }

    public function getEventId()
    {
        return $this->event_id;
    }

    public function setEventId($id)
    {
        $this->event_id = $id;
        return $this;
    }

    public function getEventTitle()
    {
        return $this->event_title;
    }

    public function setEventTitle($title)
    {
        $this->event_title = $title;
        return $this;
    }

    public function getEventDescription()
    {
        return $this->event_description;
    }

    public function setEventDescription($description)
    {
        $this->event_description = $description;
        return $this;
    }

    public function getEventExercise()
    {
        return $this->event_exercise;
    }

    public function setEventExercise($exercise)
    {
        $this->event_exercise = $exercise;
        return $this;
    }

    public function getEventIncidents()
    {
        return $this->event_incidents;
    }

    public function setEventIncidents($incidents)
    {
        $this->event_incidents = $incidents;
        return $this;
    }

    public function getEventImage()
    {
        return $this->event_image;
    }

    public function setEventImage($image)
    {
        $this->event_image = $image;
        return $this;
    }

    public function getEventOrder()
    {
        return $this->event_order;
    }

    public function setEventOrder($order)
    {
        $this->event_order = $order;
        return $this;
    }
}

File: openex-api/src/main/java/io/openex/database/model/Event.java
package io.openex.database.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.openex.helper.MonoModelDeserializer;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "events")
public class Event implements Base {
    @Id
    @Column(name = "event_id")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @JsonProperty("event_id")
    private String id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "event_exercise")
    @JsonSerialize(using = MonoModelDeserializer.class)
    @JsonProperty("event_exercise")
    private Exercise exercise;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "event_image")
    @JsonSerialize(using = MonoModelDeserializer.class)
    @JsonProperty("event_image")
    private File image;

    @Column(name = "event_title")
    @JsonProperty("event_title")
    private String title;

    @Column(name = "event_description")
    @JsonProperty("event_description")
    private String description;

    @Column(name = "event_order")
    @JsonProperty("event_order")
    private Short order;

    @OneToMany(mappedBy = "event", fetch = FetchType.EAGER)
    @Fetch(FetchMode.SUBSELECT)
    @JsonIgnore
    private List<Incident> incidents = new ArrayList<>();

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Exercise getExercise() {
        return exercise;
    }

    public void setExercise(Exercise exercise) {
        this.exercise = exercise;
    }

    public File getImage() {
        return image;
    }

    public void setImage(File image) {
        this.image = image;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Short getOrder() {
        return order;
    }

    public void setOrder(Short order) {
        this.order = order;
    }

    public List<Incident> getIncidents() {
        return incidents;
    }

    public void setIncidents(List<Incident> incidents) {
        this.incidents = incidents;
    }
}
